ps_socialfollow.php

```diff
@@ -187,11 +202,32 @@ public function getConfigFieldsValues()
     {
         $result = [];
         foreach (static::SOCIAL_NETWORKS as $social) {
-            $result['blocksocial_' . $social] = Configuration::get('BLOCKSOCIAL_' . strtoupper($social));
+            if (!empty(Configuration::get('BLOCKSOCIAL_' . strtoupper($social)))) {
+                $this->upgradeConfiguration('BLOCKSOCIAL_' . strtoupper($social));
+            }
+            foreach ($this->context->controller->getLanguages() as $lang) {
+                $id_lang = $lang['id_lang'];
+                $conf = Configuration::get('BLOCKSOCIAL_' . strtoupper($social), $id_lang);
+                $result['blocksocial_' . $social][$id_lang] = $conf;
```

Comment:
Nice-to-have: Do the same for `'blocksocial_' . $social` : have a single method to do it, use it at all need statements

Reply:
It's used only once in all the code  😄